scripts/generate_example_data.py

```diff
@@ -123,20 +113,23 @@ def generate_patient_data():
                 start_date = random_date(date_of_birth, date_of_death)
                 end_date = random_date(start_date, date_of_death)
 
-                overlap = False
-                repeat = False
-                for existing_start, existing_end in generated_dates:
-                    if start_date < existing_end and end_date > existing_start:
-                        overlap = True
-                    if start_date == existing_start and end_date == existing_end:
-                        repeat = True
-                        break
-                if not (overlap or repeat):
+                if not date_overlap_repeat(start_date, end_date, generated_dates):
                     generated_dates.append((start_date, end_date))
                     yield "practice_registrations", {"start_date": start_date, "end_date": end_date,
                                                      "practice_id": practice_pseudo_id}
                     break
 
+# check for overlap and repeat
+def date_overlap_repeat(start_date, end_date, generated_dates):
```

Comment:
Please could you add a docstring comment giving examples of inputs where this returns True and where it returns False?